flaml/model.py

```diff
@@ -1965,6 +1965,189 @@ class XGBoostLimitDepth_TS(TS_SKLearn):
     base_class = XGBoostLimitDepthEstimator
 
 
+class AGTextPredictorEstimator(BaseEstimator):
+    """
+    The class for tuning AutoGluon TextPredictor
+    """
+    def __init__(self, task="binary", **params,):
+        from autogluon.text import TextPredictor
+
+        super().__init__(task, **params)
+        self.estimator_class = TextPredictor
+
+    @classmethod
+    def search_space(cls, **params):
+        """
+        Add the possible search space configs here, e.g. 'optimization.lr'
+        reference:
+        https://auto.gluon.ai/stable/tutorials/text_prediction/customization.html#custom-hyperparameter-values
+        """
+        search_space_dict = {
+            "model.network.agg_net.mid_units": {
+                "domain": tune.choice(list(range(32, 129))),
+                "init_value": 128,
+            },
+            "optimization.lr": {
+                "domain": tune.loguniform(lower=1E-5, upper=1E-4),
+                "init_value": 1E-4,
+            },
+            "optimization.wd": {
+                "domain": tune.choice([1E-4, 1E-3, 1E-2]),
+                "init_value":1E-4,
+            },
+            "optimization.warmup_portion": {
+                "domain": tune.choice([0.1, 0.2]),
+                "init_value":0.1, 
+            },
+        }
+        return search_space_dict
+
+    def _init_fix_args(self, automl_fit_kwargs: dict=None):
+        """
+        Save the customed fix args here
+        this includes:
+            "output_dir",
+            "text_backbone": "electra_base"
+            "multimodal_fusion_strategy":"fuse_late",
+        """
+        fix_args = {}
+        FIX_ARGS_LIST = ["output_dir", "dataset_name", "label_column", "per_device_batch_size", "backend",
+                         "text_backbone", "multimodal_fusion_strategy", "num_train_epochs", "batch_size",]
+        for key, value in automl_fit_kwargs["custom_fix_args"].items():
+            assert (
+                key in FIX_ARGS_LIST
+            ), "The specified key {} is not in the argument list: output_dir, backend, label_column, dataset_name, text_backbone,\
+                multimodal_fusion_strategy, num_train_epochs, batch_size, per_device_batch_size".format(key)
+
+            fix_args[key] = value
+
+        self.fix_args = fix_args
+
+    def _init_hp_config(self, text_backbone: str, multimodal_fusion_strategy: str):

```

Comment:
Please define cfg by defining a function inside of flaml/nlp/utils.py:class AGArgs, the remove this function.

Reply:
This `_init_hp_config` is to use the AGArgs and the self.params to get the `hyperparameters`diction for the TextPredictor. If removed, still need to assemble this diction inside the `MultimodalEstimator.fit()`. Do you think it is better without this function and have this part inside the `.fit()`?